test/File/RenameTest.php

```diff
@@ -26,393 +25,328 @@
+                    'overwrite' => false,
+                    'randomize' => false,
+                ],
+                'expectedFilterResult'  => $newDirFile,
             ],
-        ]);
-
-        self::assertSame(
-            [
-                0 => [
-                    'source'    => $this->oldFile,
-                    'target'    => $this->newDir,
+            'Configured with only target and filtering file path'            => [
+                'options'               => ['target' => $newDir],
+                'input'                 => $oldFile,
+                'expectedGetFileResult' => [
+                    'source'    => '*',
+                    'target'    => $newDir,
                     'overwrite' => false,
                     'randomize' => false,
                 ],
+                'expectedFilterResult'  => $newDirFile,
             ],
-            $filter->getFile()
-        );
-        self::assertSame($this->newDirFile, $filter($this->oldFile));
-        self::assertSame('falsefile', $filter('falsefile'));
+        ];
     }
 
-    /**
-     * Test single array parameter filter by using directory only
-     */
-    public function testConstructTruncatedSourceDirectory(): void
+    public static function returnInvalidFilterInputProvider(): array
     {
-        $filter = new FileRename([
-            'target' => $this->newDir,
-        ]);
+        $oldFile = self::getTempPath() . '/' . self::TEST_FILE_NAME;
+        $newFile = self::getTempPath() . '/new_file.xml';
 
-        self::assertSame(
-            [
-                0 => [
-                    'target'    => $this->newDir,
+        return [
+            'Source file non-existent' => [
+                'options'               => $newFile,
+                'input'                 => 'non-existent-file.txt',
+                'expectedGetFileResult' => [
                     'source'    => '*',
+                    'target'    => $newFile,
                     'overwrite' => false,
                     'randomize' => false,
                 ],
+                'expectedFilterResult'  => 'non-existent-file.txt',
             ],
-            $filter->getFile()
-        );
-        self::assertSame($this->newDirFile, $filter($this->oldFile));
-        self::assertSame('falsefile', $filter('falsefile'));
+            'Only source configured'   => [
+                'options'               => ['source' => $oldFile],
+                'input'                 => $oldFile,
+                'expectedGetFileResult' => [
+                    'source'    => $oldFile,
+                    'target'    => '*',
+                    'overwrite' => false,
+                    'randomize' => false,
+                ],
+                'expectedFilterResult'  => $oldFile,
+            ],
+        ];
+    }
+
+    #[DataProvider('returnValidFilterInputProvider')]
+    #[DataProvider('returnInvalidFilterInputProvider')]
+    /** @param FilterInput $expectedFilterResult **/
```

Comment:
The `FilterInput` parameter type isn't imported anywhere AFAICT

Reply:
Thanks